cypress/e2e/audio-player/audio-player.spec.ts

```diff
@@ -0,0 +1,563 @@
+        });
+    };
+
+    // Take snapshots in modern and bubble layout, outputting log for reference/debugging
+    // Player on IRC layout should have the same layout as group layout
+    const takeSnapshots = (wrapper: string, detail: string) => {
+        // Check the status of the seek bar
+        // TODO: check if visible - currently visibility check on a narrow timeline causes an error
+        cy.get(`${wrapper}.mx_AudioPlayer_seek input[type='range']`).should("exist");
+
+        // Assert that the pause button is not rendered
+        cy.get(`${wrapper}[data-testid='play-pause-button'][aria-label='Pause']`).should("not.exist");
+
+        // Assert that the play button is rendered
+        cy.get(`${wrapper}[data-testid='play-pause-button'][aria-label='Play']`).should("exist");
+
+        // Enable group layout
+        cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.Group);
+
+        // 243 + 12px + 12px = 267px
+        // See _MediaBody.pcss and _AudioPlayer.pcss for spacing
+        cy.get(`${wrapper}.mx_MAudioBody`).percySnapshotElement(detail + " on group layout", {
+            percyCSS,
+            widths: [267],
+        });
+
+        cy.log("Took a snapshot of " + detail + " on group layout");
+
+        // Enable bubble layout
+        cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.Bubble);
+
+        // 243px + 12px + 48px = 303px
+        // See _EventBubbleTile.pcss and _AudioPlayer.pcss for spacing
+        cy.get(`${wrapper}.mx_MAudioBody`).percySnapshotElement(detail + " on bubble layout", {
+            percyCSS,
+            widths: [303],
+        });
+
+        cy.log("Took a snapshot of " + detail + " on bubble layout");
+
+        // Reset the layout
+        cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.Group);
+    };
+
+    beforeEach(() => {
+        cy.startHomeserver("default").then((data) => {
+            homeserver = data;
+            cy.initTestUser(homeserver, TEST_USER).then(() =>
+                cy.createRoom({}).then((_roomId) => {
+                    roomId = _roomId;
+                }),
+            );
+        });
+
+        cy.injectAxe();
+    });
+
+    afterEach(() => {
+        cy.stopHomeserver(homeserver);
+    });
+
+    it("should render player on every layout", () => {
+        visitRoom();
+
+        // Upload one second audio file with a long file name
+        uploadFile("cypress/fixtures/1sec-long-name-audio-file.ogg");
+
+        // Scroll to the bottom to make the audio player visible for Percy tests
+        cy.get(".mx_MainSplit .mx_ScrollPanel").scrollTo("bottom");
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            checkPlayerFilenameLong();
+
+            // Click the timestamp to highlight the event tile on which player is rendered
+            cy.get(".mx_EventTile .mx_MessageTimestamp").click();
+            cy.get(".mx_EventTile.mx_EventTile_selected .mx_MAudioBody").should("be.visible");
```

Comment:
could we move this inside `takeSnapshots`? We seem to do it quite often, and it would make `takeSnapshots` consistent with `takeSnapshotReply`, `takeSnapshotTimeline`, etc.

Reply:
I think it is possible. Let me check on Cypress locally in case!

Reply:
With this commit https://github.com/matrix-org/matrix-react-sdk/pull/10441/commits/725d020c533291c5c1ccddec033ef83cfbd37d7c I created a function to check player's visibility, removing obsolete assertions.

The function is added to `takeSnapshots`, `takeSnapshotReply`, `takeSnapshotReplyChain`, and `takeSnapshotTimeline`.